examples/helloworld_cloudevents/helloworld.go

```diff
@@ -0,0 +1,95 @@
+package helloworld_cloudevents
+
+import (
+	"embed"
+	"fmt"
+	"os"
+
+	"k8s.io/client-go/rest"
+	"open-cluster-management.io/addon-framework/examples/rbac"
+	"open-cluster-management.io/addon-framework/pkg/addonfactory"
+	"open-cluster-management.io/addon-framework/pkg/agent"
+	"open-cluster-management.io/addon-framework/pkg/utils"
+	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	workapiv1 "open-cluster-management.io/api/work/v1"
+)
+
+const (
+	DefaultImage          = "quay.io/open-cluster-management/addon-examples:latest"
+	AddonName             = "helloworldcloudevents"
+	InstallationNamespace = "open-cluster-management-agent-addon"
+)
+
+//go:embed manifests
+//go:embed manifests/templates
+var FS embed.FS
+
+func NewRegistrationOption(kubeConfig *rest.Config, addonName, agentName string) *agent.RegistrationOption {
+	return &agent.RegistrationOption{
+		CSRConfigurations: agent.KubeClientSignerConfigurations(addonName, agentName),
+		CSRApproveCheck:   utils.DefaultCSRApprover(agentName),
+		PermissionConfig:  rbac.AddonRBAC(kubeConfig),
+	}
+}
+
+func GetDefaultValues(cluster *clusterv1.ManagedCluster,
+	addon *addonapiv1alpha1.ManagedClusterAddOn) (addonfactory.Values, error) {
+
+	image := os.Getenv("EXAMPLE_IMAGE_NAME")
+	if len(image) == 0 {
+		image = DefaultImage
+	}
+
+	manifestConfig := struct {
+		KubeConfigSecret string
+		ClusterName      string
+		Image            string
+	}{
+		KubeConfigSecret: fmt.Sprintf("%s-hub-kubeconfig", addon.Name),
+		ClusterName:      cluster.Name,
+		Image:            image,
+	}
+
+	return addonfactory.StructToValues(manifestConfig), nil
+}
+
+func AgentHealthProber() *agent.HealthProber {
```

Comment:
is there any specific reason that we need to use the work prober to check the health for cloud events driver addons?

Reply:
the cloudevents driver added in this PR is only used for addon agent deploy, no other difference from the current "kube" driver. Therefore, it use similar prober to make sure addon agent is deployed successfully on managed cluster as "kube" driver. Make sense?